src/diffusers/models/modeling_utils.py

```diff
@@ -89,12 +89,12 @@ def find_tensor_attributes(module: torch.nn.Module) -> List[Tuple[str, Tensor]]:
         return first_tuple[1].dtype
 
 
-def load_state_dict(checkpoint_file: Union[str, os.PathLike]):
+def load_state_dict(checkpoint_file: Union[str, os.PathLike], variant: Optional[str] = None):
```

Comment:
we could directly check the extension of `checkpoint_file` to know whether or not to load with safetensors. That way we wouldn't have to pass in variant